src/init/init.c

```diff
@@ -30,6 +33,20 @@ struct sof *sof_get(void)
 	return &sof;
 }
 
+struct sof_core_context **sof_get_core_context(void)
+{
+	struct core_context *ctx = (struct core_context *)cpu_read_threadptr();
+
+	return &ctx->sof;
+}
```

Comment:
This function and related may be worth separating into another commit.

Reply:
Also, what happens when slave cores get turned off? Is it possible for that to happen outside the situation that the DSP itself is fully powered off?

Reply:
I don't know exactly, I'm not familiar with it.
However, judging from the code I investigated, the core context is simply gone in a case like that. So, after the slave core gets turned on again, it allocates the context all over again.

Reply:
Alright, that _may_ be a leak unless you use the SYSTEM heap for that core (I'm not sure, didn't look carefully enough to check this specifically). I'll leave it up to you to look at that though and say it's okay from my end. Thanks for the explanations.

Reply:
Yeah, the system heap is used for that.